Vagrantfile

```diff
@@ -29,6 +29,15 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   else
     config.vm.network :private_network, ip: vconfig['vagrant_ip']
   end
+
+  if vconfig['vagrant_enable_public_network']
```

Comment:
Let's simplify this a bit by just using `vagrant_public_ip`. If it has a value, and the value is 0.0.0.0, do the first condition, else if it has a value, do the second condition:

```ruby
  if !vconfig['vagrant_public_ip'].empty? && vconfig['vagrant_public_ip'] == "0.0.0.0"
    config.vm.network :public_network
  elsif !vconfig['vagrant_public_ip'].empty?
    config.vm.network :public_network, ip: vconfig['vagrant_public_ip']
  end
```